server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorTests.java

```diff
@@ -1377,6 +1382,56 @@ public void testSortingWithNestedAggregations() throws IOException {
         }
     }
 
+    public void testManySegmentsStillSingleton() throws IOException {
```

Comment:
I'm a little unhappy that this test doesn't run through `AggregatorTestCase.testCase`, or at least use `AggregatorTestCase.searchAndReduce`.   I think that's because you want to assert on the debug info which we don't currently make easy.  If that's correct, I don't think we need to address it here, but I would like to leave a todo and maybe a ticket.

Reply:
Done!